### PR TITLE
Automatically focus first input during restore seed pagination.

### LINKED ISSUE #729: Autofocus is broken starting on the 2nd page of seed restore

### PR DESCRIPTION
**Pull request by valentinewallace**
Closes #729.

### CODE REVIEW

File: src/component/input.js [hunk 1]
```diff
@@ -20,19 +20,32 @@ const baseStyles = StyleSheet.create({
...
-    underlineColorAndroid="rgba(0,0,0,0)"
-    placeholderTextColor={color.greyPlaceholder}
-    {...props}
-  />
-);
+export class TextInput extends Component {
+  componentDidUpdate() {
+    const { autoFocus, value } = this.props;
+    value.length === 0 && autoFocus && this._input.focus();
+  }
```
Comment (tanx) [inline src/component/input.js]: Cleaned this up a bit in https://github.com/lightninglabs/lightning-app/pull/781/commits/f8010e922c04e7e7adbe40be6b72a82286e7ca02

**Comment by valentinewallace** [comment]
I tried flipping `autoFocus` from true to false in `view/seed-verify`, but that still didn't refocus the correct input for me. This solution seems work across screens that use autofocus, but let me know if it doesn't work for you.

It's a bit hacky still :( maybe we should revisit this in a bit?

**Comment by tanx** [comment]
@valentinewallace is this PR still relevant with the recent patch to textinput focus?

**Comment by valentinewallace** [comment]
@tanx Yep, just tested and recovery input autofocus is still a bug.

File: src/view/restore-seed.js [hunk 2]
```diff
@@ -52,7 +52,7 @@ const RestoreSeedView = ({ store, wallet }) => (
               wallet.setRestoreSeed({ word, index: seedIndex - 1 })
             }
             key={i}
-            autoFocus={i === 0}
+            autoFocus={i === 0 && store.wallet.autofocusRestoreInput}
```
Comment (tanx) [inline src/view/restore-seed.js]: @valentinewallace Not really sure I understand ... why does `store.wallet.autofocusRestoreInput` need to be false for the `setRestoreSeed()` case?

Comment (valentinewallace) [reply src/view/restore-seed.js]: Disclaimer: I don't fully know why this solution works.

It needs `autofocusRestoreInput` to flip from false to true when the user turns the page, or else the bug persists. I decided to set it to false in `setRestoreSeed()` but there may be a better place.

**Comment by valentinewallace** [comment]
Replaced by #1106.

**Pull request closed without merging**